### PR TITLE
Feature/goreleaser

### PR DESCRIPTION
**Pull request by evalsocket**
# TL;DR
Added change for publishing flytectl binary in brew and scoop

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
-  Added goreleaser change for brew and scoop
-  Added workflow for  goreleaser 
-  Added `install.sh` script for darwin and Linux

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -83,21 +76,14 @@ brews:
...
-    # Gitea is not supported yet, but the support coming
+    tap:
+      owner: flyteorg
+      name: flytectl
 
     # Template for the url which is determined by the given Token (github or gitlab)
     # Default for github is "https://github.com/<repo_owner>/<repo_name>/releases/download/{{ .Tag }}/{{ .ArtifactName }}"
     # Default for gitlab is "https://gitlab.com/<repo_owner>/<repo_name>/uploads/{{ .ArtifactUploadHash }}/{{ .ArtifactName }}"
-    url_template: "http://github.com/lyft/flytectl/releases/{{ .Tag }}/{{ .ArtifactName }}"
+    url_template: "http://github.com/flyte/flytectl/releases/{{ .Tag }}/{{ .ArtifactName }}"
```
Comment (EngHabu) [inline .goreleaser.yml:86]:
```suggestion
    url_template: "http://github.com/flyteorg/flytectl/releases/{{ .Tag }}/{{ .ArtifactName }}"
```

**Review by EngHabu — DISMISSED** [review]

File: .goreleaser.yml [hunk 2]
```diff
@@ -115,17 +101,13 @@ brews:
       name: goreleaserbot
       email: goreleaser@carlosbecker.com
```
Comment (kumare3) [inline .goreleaser.yml:102]: you should to change the email.
Lets use flytebot

File: .goreleaser.yml [hunk 3]
```diff
@@ -115,17 +101,13 @@ brews:
...
-    folder: Formula
-
     # Caveats for the user of your binary.
     # Default is empty.
     caveats: "How to use this binary"
 
     # Your app's homepage.
     # Default is empty.
-    homepage: "https://godoc.org/github.com/lyft/flytectl"
+    homepage: "https://godoc.org/github.com/flyteorg/flytectl"
```
Comment (kumare3) [inline .goreleaser.yml:110]: can we change this to https://flytectl.readthedocs.io/en/latest/

**Comment by evalsocket** [comment]
@kumare3 @EngHabu i will fix lint in my other pr

**Comment by kumare3** [comment]
@evalsocket do you want to push your other change?

**Comment by evalsocket** [comment]
Yes....i am planing to resolve all lint in same PR.

**Comment by evalsocket** [comment]
Yes

**Comment by evalsocket** [comment]
@kumare3 @EngHabu you can review

File: cmd/get/execution.go [hunk 4]
```diff
@@ -49,12 +50,12 @@ Usage
 )
 
 var executionColumns = []printer.Column{
-	{"Name", "$.id.name"},
-	{"Workflow Name", "$.closure.workflowId.name"},
-	{"Type", "$.closure.workflowId.resourceType"},
-	{"Phase", "$.closure.phase"},
-	{"Started", "$.closure.startedAt"},
-	{"Elapsed Time", "$.closure.duration"},
+	{Header: "Name", JSONPath: "$.id.name"},
```
Comment (kumare3) [inline cmd/get/execution.go:53]: Why this change?

File: cmd/register/register_util.go [hunk 5]
```diff
@@ -127,27 +128,37 @@ func hydrateNode(node *core.Node) error {
 			launchPlanNodeReference := workflowNodeWrapper.WorkflowNode.Reference.(*core.WorkflowNode_LaunchplanRef)
 			hydrateIdentifier(launchPlanNodeReference.LaunchplanRef)
 		default:
-			errors.New(fmt.Sprintf("Unknown type %T", workflowNodeWrapper.WorkflowNode.Reference))
+			fmt.Printf("Unknown type %T", workflowNodeWrapper.WorkflowNode.Reference)
```
Comment (kumare3) [inline cmd/register/register_util.go:131]: ohh this error should be returned!

Comment (kumare3) [reply cmd/register/register_util.go:131]: @pmahindrakar-oss FYI

File: cmd/register/register_util.go [hunk 6]
```diff
@@ -127,27 +128,37 @@ func hydrateNode(node *core.Node) error {
...
 		}
 	case *core.Node_BranchNode:
 		branchNodeWrapper := targetNode.(*core.Node_BranchNode)
-		hydrateNode(branchNodeWrapper.BranchNode.IfElse.Case.ThenNode)
+		if err := hydrateNode(branchNodeWrapper.BranchNode.IfElse.Case.ThenNode); err != nil {
+			// TODO Write Error Log
+			fmt.Println("failed to hydrateNode")
```
Comment (kumare3) [inline cmd/register/register_util.go:137]: I think we should return this error - not just log it - @pmahindrakar-oss, we need to fix this

File: cmd/root.go [hunk 7]
```diff
@@ -43,7 +44,7 @@ func newRootCmd() *cobra.Command {
 	rootCmd.AddCommand(versionCmd)
 	rootCmd.AddCommand(get.CreateGetCommand())
 	rootCmd.AddCommand(update.CreateUpdateCommand())
-	rootCmd.AddCommand(register.RegisterCommand())
+	rootCmd.AddCommand(register.FlyteRegisterCommand())
```
Comment (kumare3) [inline cmd/root.go:47]: Why not register command? I guess because it has a weird meaning.
What about `RemoteRegisterCommand`

**Comment by kumare3** [comment]
Thank you @evalsocket for doing these changes. I would like @pmahindrakar-oss to take a look once and probably some linter errors are in the new code that he introduced.. Once that +1

File: cmd/register/register_util.go (hunk 6 above)
Comment (pmahindrakar-oss) [reply cmd/register/register_util.go:137]: Returning an error in earlier code would cause panic in main.go
Now that this is being changed to been logged to stdout
in this change https://github.com/flyteorg/flytectl/pull/17/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R10
then we should be returning this error all the way up. 

Will check and fix after the above mr gets merged.

File: cmd/update/project.go [hunk 8]
```diff
@@ -80,7 +81,7 @@ func updateProjectsFunc(ctx context.Context, args []string, cmdCtx cmdCore.Comma
 	archiveProject := projectConfig.ArchiveProject
 	activateProject := projectConfig.ActivateProject
 	if activateProject == archiveProject {
-		fmt.Printf(errInvalidUpdate)
+		fmt.Println(errInvalidUpdate)
```
Comment (pmahindrakar-oss) [inline cmd/update/project.go:84]: Instead of printing , can we do the same here aswell , return the error and let the main.go handle it by logging it

**Comment by pmahindrakar-oss** [comment]
Thanks @evalsocket for taking care of the linter issues in the recently changed files. Changes look good to me

File: cmd/get/execution.go (hunk 4 above)
Comment (evalsocket) [reply cmd/get/execution.go:53]: lint error

**Comment by evalsocket** [comment]
@kumare3 @pmahindrakar-oss you can review the pr. It is ready

**Review by kumare3 — APPROVED** [review]
